pkg/apis/core/validation/validation.go

```diff
@@ -5005,6 +5005,16 @@ func ValidateSecretUpdate(newSecret, oldSecret *core.Secret) field.ErrorList {
 	}
 
 	allErrs = append(allErrs, ValidateImmutableField(newSecret.Type, oldSecret.Type, field.NewPath("type"))...)
+	if oldSecret.Immutable != nil && *oldSecret.Immutable {
+		if !reflect.DeepEqual(newSecret.Immutable, oldSecret.Immutable) {
```

Comment:
Don't use reflect for this - it's just a bool right?  Whan I see reflect, I assume something tricky and subtle is happening...is it?

Reply:
yes, IMO

Reply:
Opened https://github.com/kubernetes/kubernetes/pull/87602 for the cleanup